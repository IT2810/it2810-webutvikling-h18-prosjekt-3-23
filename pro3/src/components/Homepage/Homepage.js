import React from 'react';
import { StyleSheet, Text, View , Image, AsyncStorage, TouchableHighlight} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import ScoreManager from '../../utils/ScoreManager';

class Homepage extends React.Component {
    constructor(props){
        super(props)

        this.state = {
            taskScore : 0,
            appScore : 0
        }

        //Updates the state of taskScore and appScore everytime the HomePage is in focus (in the navigation-menu)
        this.props.navigation.addListener("didFocus", () => {this.retrieveTaskScoreAsync()})
        this.props.navigation.addListener("didFocus", () => {this.retrieveAppScoreAsync()})
    }


    componentDidMount(){
        this.retrieveTaskScoreAsync()
        this.retrieveAppScoreAsync()
    }

    //Retrieves the taskScore
    retrieveTaskScoreAsync = async () => {
        let taskScore = await AsyncStorage.getItem("TASKSCORE");
        this.setState({taskScore : taskScore});
        }

    //Retrieves the appointments score
    retrieveAppScoreAsync = async () => {
        let appScore = await AsyncStorage.getItem("APPSCORE");
        this.setState({appScore : appScore})
    }

    //Resets the taskScore back to zero
    clearTaskScore() {
        let score = 0
        this.setState({taskScore : score})
        ScoreManager.saveTaskScore(score.toString())
    }



    render() {
        return (
            <View style={styles.container}>
                <View style={styles.imageHolder}> 
                    <Image style={{height:180, width:220}} source={require('./logo.png')} />
                </View>
                
               <View style={styles.hr}/>
                    <View style={styles.scoreHolder}>
                        <Text style={styles.scoreText}>TodoÂ´s completed:  {this.state.taskScore}</Text>
                        <TouchableHighlight onPress={() => this.clearTaskScore()}>
                            <Icon name="times" size={24} color="#607D8B"/>
                        </TouchableHighlight>
                    </View>

                <View style={styles.hr}/>
                    <View style={styles.scoreHolder}>
                        <Text style={styles.scoreText}>Appointments made: {this.state.appScore}</Text>
                    </View>
                <View style={styles.hr}/>

            </View>
        );
    }
}

export default Homepage;


const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#37474f',
        alignItems: 'center',
        justifyContent: 'center',
    },
    scoreText: {
        color: '#CFD8DC',
        fontSize: 24
    },
    scoreHolder : {
        flexDirection:"row",
        justifyContent:"space-between",
        alignItems: "center",
        width: "70%"
    },
    hr: {
        height: 1,
        backgroundColor: "#607D8B",
        width: "100%",
        margin: 20
    },
    imageHolder : {
        margin: 25
    }

});
