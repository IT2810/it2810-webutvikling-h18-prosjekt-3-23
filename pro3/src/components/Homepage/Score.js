import React from 'react';
import { StyleSheet, Text, View } from 'react-native';



class Score extends React.Component {

    render() {
        return (
            <View>
                <Text style={styles.scoreText}>Score</Text>
            </View>
        );
    }
}


export default Score;

const styles = StyleSheet.create({
    scoreText: {
        fontFamily: 'HelveticaNeue-UltraLight',
        color: '#eceff1',
        fontSize: 30,
    }
})
