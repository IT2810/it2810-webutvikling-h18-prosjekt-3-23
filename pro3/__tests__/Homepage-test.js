import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Homepage from '../src/components/Homepage/Homepage';
import MockAsyncStorage from 'mock-async-storage';
import { AsyncStorage as storage } from 'react-native';

const navigationEx = { addListener: jest.fn() };

//Implements mocks
const mock = () => {
    const mockImpl = new MockAsyncStorage()
    jest.mock('AsyncStorage', () => mockImpl)
};

it('should render corrrectly', async () => {
    mock();
    const tree = renderer.create(<Homepage navigation={navigationEx}/>).toJSON();
    expect (tree).toMatchSnapshot();
    jest.unmock('AsyncStorage');
});

it('Mock Async Storage working', async () => {
    await storage.setItem('TASKSCORE', '5')
    const value = await storage.getItem('TASKSCORE')
    expect(value).toBe('5')
});