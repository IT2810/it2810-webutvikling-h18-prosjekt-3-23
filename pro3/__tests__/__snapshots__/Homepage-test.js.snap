// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render corrrectly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#263238",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 180,
        "width": 220,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 20,
        "width": "70%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eceff1",
          "fontSize": 24,
        }
      }
    >
      Todo's completed: 
      0
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#607D8B",
        "height": 1,
        "marginTop": 5,
        "width": "70%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 20,
        "width": "70%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eceff1",
          "fontSize": 24,
        }
      }
    >
      Appointments made: 
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#607D8B",
        "height": 1,
        "marginTop": 5,
        "width": "70%",
      }
    }
  />
</View>
`;
